src/typing/index.ts

```diff
@@ -0,0 +1,80 @@
+export type TUser = {
+  data: TUserProps;
+};
+
+export type TUserProps = {
+  id: number;
+  first_name: string;
+  second_name: string;
+  display_name: string;
+  login: string;
+  email: string;
+  phone: string;
+  avatar: string;
+};
+
+export type TMessage = {
+  chat_id: number;
+  time: string;
+  type: string;
+  user_id: string;
+  content: string;
+  file?: {
+    id: number;
+    user_id: number;
+    path: string;
+    filename: string;
+    content_type: string;
+    content_size: number;
+    upload_date: string;
+  };
+};
+
+export type TChatInfo = {
+  id: number;
+  title: string;
+  avatar: string;
+  unread_count: number;
+  last_message: {
+    user: TUser;
+    time: string;
+    content: string;
+  };
+};
+
+export type TChangeProfileData = {
+  first_name: string;
+  second_name: string;
+  display_name: string;
+  login: string;
+  email: string;
+  phone: string;
+};
+
+export type TChangePassword = {
+  oldPassword: string;
+  newPassword: string;
+};
+
+export type TSignUpData = {
+  first_name: string;
+  second_name: string;
+  login: string;
+  email: string;
+  password: string;
+  phone: string;
+};
+
+export type TSignInData = {
+  login: string;
+  password: string;
+};
+
+export type TProps = {
+  events?: Record<string, (e?: Event) => void>;
```

Comment:
```suggestion
  events?: Record<string, (e: Event) => void>;
```

Если можно обойтись без nullable-типов, надо обходиться без nullable-типов

Reply:
Если убрать тот optional operator, то из кода сразу исчезает [optional chaining operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html), что сразу повышает безопасность кода